src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,124 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private static RegistrationServiceImpl registrationService;
+    private StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    public User createUser(String login, String password, int age) {
+        User user = new User();
+        user.setLogin(login);
+        user.setPassword(password);
+        user.setAge(age);
+        return user;
+    }
+
+    @Test
+    void register_Check_Age_Ok() {
+        User user1 = createUser("john1993", "123456", 20);
+        User user2 = createUser("jack1993", "123456", 18);
+        registrationService.register(user1);
+        registrationService.register(user2);
+        assertEquals(storageDao.get("john1993"), user1);
+        assertEquals(storageDao.get("jack1993"), user2);
+    }
+
+    @Test
+    void register_Check_Age_NotOk() {
+        User user1 = createUser("qwerty123", "123456", 15);
+        User user2 = createUser("wanted567", "123456", 12);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user2);
+        });
+    }
+
+    @Test
+    void register_Check_LoginNull_NotOk() {
```

Comment:
fix naming everywhere
```suggestion
    void register_nullLogin_notOk() {
```